PGM/src/main/java/tc/oc/pgm/shop/PurchaseTracker.java

```diff
@@ -0,0 +1,96 @@
+package tc.oc.pgm.shop;
+
+import org.bukkit.event.Listener;
+import tc.oc.api.docs.User;
+import tc.oc.pgm.filters.Filter;
+import tc.oc.pgm.match.MatchPlayer;
+import tc.oc.pgm.match.Party;
+import tc.oc.pgm.shop.purchasable.Purchasable;
+import tc.oc.pgm.shop.strategy.PaymentStrategy;
+import tc.oc.pgm.shop.strategy.PlayerPaymentStrategy;
+
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Class which is used to track all {@link PaymentStrategy PaymentStrategies}.*
+ */
+public interface PurchaseTracker extends Listener {
+    /**
+     * Start a strategy for an individual player.
+     * @param player who initiated the strategy.
+     * @param purchasable that the strategy is working towards.
+     */
+    void startIndividualStrategy(MatchPlayer player, Purchasable purchasable);
+
+    /**
+     * Start a strategy for a party.
+     * @param party that the strategy is based upon.
+     * @param player who initiated the strategy.
+     * @param purchasable that the strategy is working towards.
+     */
+    void startPartyStrategy(Party party, MatchPlayer player, Purchasable purchasable);
+
+    /**
+     * Start a strategy that anyone in the {@link tc.oc.pgm.match.Match} can contribute to.
+     * @param player who initiated the strategy.
+     * @param purchasable that the strategy is working towards.
+     */
+    void startGlobalStrategy(Purchasable purchasable, MatchPlayer player);
+
+    /**
+     * {@link #startIndividualStrategy(MatchPlayer, Purchasable)} with a contribution filter.
+     */
+    void startIndividualStrategy(MatchPlayer player, Purchasable purchasable, Filter filter);
+
+    /**
+     * {@link #startPartyStrategy(Party, MatchPlayer, Purchasable)} with a contribution filter.
+     */
+    void startPartyStrategy(Party party, MatchPlayer player, Purchasable purchasable, Filter filter);
+
+    /**
+     * {@link #startGlobalStrategy(Purchasable, MatchPlayer)} with a contribution filter.
+     */
+    void startGlobalStrategy(Purchasable purchasable, MatchPlayer player, Filter filter);
+
+    /**
+     * Called when a {@link MatchPlayer} contributes to a {@link PaymentStrategy} after all preconditions are met.
+     * @param strategy that is being contributed to
+     * @param player that is contributing
+     */
+    void contribute(PaymentStrategy strategy, MatchPlayer player);
+
+    /**
+     * Get an individual strategy that is ongoing for the supplied {@link Purchasable}.
+     * @param player that owns the strategy.
+     * @param purchasable that the strategy is working towards.
+     * @return
+     */
+    Optional<PaymentStrategy> getOngoingStrategy(MatchPlayer player, Purchasable purchasable);
```

Comment:
The idea with this is that it could be misleading to implementers since you will never get a strategy that has completed.